pkg/controller/backup_controller.go

```diff
@@ -256,7 +251,7 @@ func patchBackup(original, updated *api.Backup, client arkv1client.BackupsGetter
 	return res, nil
 }
 
-func (c *backupController) getLocationAndValidate(itm *api.Backup, defaultBackupLocation string) (*api.BackupStorageLocation, []string) {
+func (c *backupController) getLocationAndValidate(itm *pkgbackup.Request, defaultBackupLocation string) []string {
```

Comment:
I think these methods might need to be renamed for the updated behavior. By its name, I'm not expecting the side-effects to be applied to the Request.